Sources/SwiftKafka/KafkaProducer.swift

```diff
@@ -208,19 +208,16 @@ public final class KafkaProducer: Service, Sendable {
             let nextAction = self.stateMachine.withLockedValue { $0.nextPollLoopAction() }
             switch nextAction {
             case .pollWithoutYield(let client):
-                // Drop any incoming acknowledgments
+                // Drop any incoming events
                 let _ = client.eventPoll()
             case .pollAndYield(let client, let source):
                 let events = client.eventPoll()
-                for event in events {
-                    switch event {
-                    case .deliveryReport(let results):
+                events
+                    .map(KafkaProducerEvent.fromKafkaEvent(_:))
+                    .forEach {
```

Comment:
Can we keep the for loop here please and just do a switch. The problem with your code is that we actually iterate the events twice. We could use `lazy` but that sometimes fails to optimise so let's rather spell it out.